Library/Homebrew/formula_installer.rb

```diff
@@ -98,8 +98,13 @@ def pour_bottle?(install_bottle_options = { :warn=>false })
 
   def install_bottle_for?(dep, build)
     return pour_bottle? if dep == formula
-    return false if build_from_source?
-    return false unless dep.bottle && dep.pour_bottle?
+    # We want to behave different with `--build-from-source` and
+    # ENV["HOMEBREW_BUILD_FROM_SOURCE"] because the latter implies you want
+    # everything built from source and the prior that you want just the
+    # formulae you've requested built from source.
+    return false if ENV["HOMEBREW_BUILD_FROM_SOURCE"]
+    return false unless dep.bottle
+    return false unless dep.pour_bottle?
```

Comment:
Same here, `return false unless dep.bottle && dep.pour_bottle?` is the same as

```ruby
return false if !(dep.bottle && dep.pour_bottle?)
```

is the same as

```ruby
return false if !dep.bottle || !dep.pour_bottle?
```

and this can be split into:

```ruby
return false unless dep.bottle
return false unless dep.pour_bottle?
```

while making sure to still evaluate them in the right order.